app/queues/common/expiration-request.server.ts

```diff
@@ -0,0 +1,101 @@
+import { Worker, Queue, UnrecoverableError } from 'bullmq';
+
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { prisma } from '~/db.server';
+
+import { addNotification } from '../notifications/notifications.server';
+import { deleteDnsRequest } from '../dns/delete-record-flow.server';
+import { RecordType } from '../notifications/expiration-notification.server';
+
+import type { ExpirationStatusPayload as ExpirationRequestPayload } from '../notifications/expiration-notification.server';
+
+const { EXPIRATION_REPEAT_FREQUENCY_MS, JOB_REMOVAL_FREQUENCY_MS } = process.env;
+const JOB_REMOVAL_INTERVAL = 7 * 24 * 60 * 60 * 1000;
+declare global {
+  var __expiration_request_init__: boolean;
+}
+// queue name
+const expirationRequestQueueName = 'expiration-request';
+
+// query to fetch expired records from database based on type
+const getExpiredRecords = (type: RecordType) => {
+  switch (type) {
+    case RecordType.DnsRecord:
+      return prisma.record.findMany({
+        where: {
+          expiresAt: {
+            lt: new Date(),
+          },
+        },
+        include: {
+          user: true,
+        },
+      });
+    case RecordType.Certificate:
+      logger.info(`TODO: return expired certificates from the db`);
+      break;
+    default:
+      throw new Error(`Invalid recordType: ${type}`);
+  }
+};
+
+// queue initialization
+const expirationRequestQueue = new Queue<ExpirationRequestPayload>(expirationRequestQueueName, {
+  connection: redis,
+});
+
+export function addExpirationRequest(type: RecordType) {
+  const jobName = `expirationRequestQueueName-${type}`;
+  return expirationRequestQueue.add(
+    jobName,
+    { type },
+    {
+      repeat: { every: Number(EXPIRATION_REPEAT_FREQUENCY_MS) || 24 * 60 * 60 * 1000 },
```

Comment:
I'd make these mandatory and throw an exception if they are missing

Reply:
Do we still need to throw an exception if we provide a hard-coded default value?

Reply:
We already have too many config options to maintain.  If we can provide a sane default and override when we want to change, I'd prefer that.